packages/browser/src/ClientAuthentication.spec.ts

```diff
@@ -97,8 +107,44 @@ describe("ClientAuthentication", () => {
     );
   }
 
+  beforeEach(() => {
+    jest.clearAllMocks();
+
+    Object.defineProperty(window, "location", {
+      writable: true,
+      value: { assign: jest.fn() },
+    });
+  });
+
   describe("login", () => {
     const mockEmitter = new EventEmitter();
+    // TODO: add tests for events & errors
+
+    it("calls login, and uses the window.location.href for the redirect if no redirectUrl is set", async () => {
+      // Set a current window location which will be the expected URL to be
+      // redirected back to, since we don't pass redirectUrl:
+      window.location.href = "https://coolapp.test/some/redirect";
+
+      const clientAuthn = getClientAuthentication();
+      await clientAuthn.login(
+        {
+          sessionId: "mySession",
+          tokenType: "DPoP",
+          clientId: "coolApp",
+          oidcIssuer: "https://idp.com",
```

Comment:
As this is a real url, let's stick with `test.url` here and elsewhere.

Reply:
This is probably a much wider issue to resolve? I think we should add a ticket for this tech debt across the SDKs